pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -143,7 +143,6 @@ func (c *Controller) Services() ([]*model.Service, error) {
 					sp.ClusterVIPs = make(map[string]string)
 				}
 				sp.ClusterVIPs[r.ClusterID] = s.Address
-				smap[s.Hostname] = sp
```

Comment:
It is necessary for multiple clusters, because there may be same service hostname in multiple clusters.

Reply:
It's just the if you check the logic before this one then you'll see that it has already assigned the map value so this is a 2nd time.

Reply:
Sorry, I do not find other places to set the map.

Reply:
L136 sets it if it wasn't found. If it's found then you don't need to set it again as it is pointer value.

Reply:
@supereagle Am I correct or do I miss something that you see and I don't?

Reply:
oh, sorry. I mistakenly think that you are changing L136.

LGTM 😆